lib/segment/src/segment_constructor/segment_constructor_base.rs

```diff
@@ -1088,3 +1028,97 @@ pub fn migrate_rocksdb_dense_vector_storage_to_mmap(
 
     Ok(new_storage)
 }
+
+/// Migrate a RocksDB based multi dense vector storage into the mmap format
+///
+/// Creates a new mutable in-memory vector storage on top of memory maps, and copies all vectors
+/// from the RocksDB based storage into it. The persisted RocksDB data is deleted so that only the
+/// new vector storage will be loaded next time. The new vector storage is returned.
+#[cfg(feature = "rocksdb")]
+pub fn migrate_rocksdb_multi_dense_vector_storage_to_mmap(
+    old_storage: VectorStorageEnum,
+    dim: usize,
+    multi_vector_config: MultiVectorConfig,
+    vector_storage_path: &Path,
+) -> OperationResult<VectorStorageEnum> {
+    use crate::vector_storage::multi_dense::appendable_mmap_multi_dense_vector_storage::find_storage_files;
+
+    log::info!(
+        "Migrating {} points in multi dense vector storage from RocksDB into new format",
+        old_storage.total_vector_count(),
+    );
+
+    fn migrate(
+        old_storage: &VectorStorageEnum,
+        dim: usize,
+        multi_vector_config: MultiVectorConfig,
+        vector_storage_path: &Path,
+    ) -> OperationResult<VectorStorageEnum> {
```

Comment:
nit: this can be a bit less verbose with a closure, as arguments can be safely captured
```suggestion
    let migrate = || {
```

Reply:
I would prefer explicit, tbh